cg/apps/downsample/utils.py

```diff
@@ -0,0 +1,57 @@
+"""Utility functions for the downsampledata module."""
+import logging
+from pathlib import Path
+
+from cg.apps.housekeeper.hk import HousekeeperAPI
+from cg.constants import SequencingFileTag
+from cg.utils.files import get_files_matching_pattern
+
+LOG = logging.getLogger(__name__)
+
+
+def store_downsampled_sample_bundle(
+    fastq_file_output_directory: str, sample_id: str, housekeeper_api: HousekeeperAPI
+) -> None:
+    """
+    Add a downsampled sample bundle to housekeeper and include the fastq files.
+    Raises:
+        FileExistsError
+    """
+    if not Path(fastq_file_output_directory).exists():
+        raise FileExistsError(f"Cannot find: {fastq_file_output_directory}")
+    create_downsampled_sample_bundle(sample_id=sample_id, housekeeper_api=housekeeper_api)
+    store_downsampled_fastq_files_from_dir(
+        sample_id=sample_id,
+        housekeeper_api=housekeeper_api,
+        fastq_file_output_directory=fastq_file_output_directory,
+    )
+
+
+def store_downsampled_fastq_files_from_dir(
+    fastq_file_output_directory: str, sample_id: str, housekeeper_api: HousekeeperAPI
+) -> None:
+    """
+    Add downsampled fastq files from the given directory to housekeeper.
+    Raises:
+        FileNotFoundError
+    """
+    fastq_file_paths: list[Path] = get_files_matching_pattern(
+        directory=Path(fastq_file_output_directory),
+        pattern=f"*{sample_id.split(sep='_')[0]}*.{SequencingFileTag.FASTQ}.gz",
+    )
+    if not fastq_file_paths:
+        raise FileNotFoundError(
+            f"No fastq files found with pattern: *{sample_id}*.{SequencingFileTag.FASTQ}.gz"
+        )
```

Comment:
Since we dont have any tracking of the downsampling - what hinders the user from running this command mid downsampling? Can we not create a very primitive way of finding if the downsampling is done or ongoing?

Reply:
Nothing hinders them. We would need to fetch the sbatch job and see if its pending. I could not see functionality for this in the SlurmAPI. 

I added a note to the store command that one should check that the downsample command is finished before running it for now.